drivers/storage/portworx/util/util.go

```diff
@@ -150,6 +150,15 @@ const (
 	// EnvKeyMarketplaceName env var for the name of the source marketplace
 	EnvKeyMarketplaceName = "MARKETPLACE_NAME"
 
+	// EnvKeyAutopilotCASecretName env var for the name of the k8s secret containing the CA cert needed to connect to portworx when TLS is enabled
+	EnvKeyAutopilotCASecretName = "PX_CA_CERT_SECRET"
+	// EnvKeyAutopilotCASecretKey env var for the name of the key in the k8s secret which will retrieve the CA cert needed to connect to portworx when TLS is enabled
+	EnvKeyAutopilotCASecretKey = "PX_CA_CERT_SECRET_KEY"
+	// DefaultCASecretName is the default value for EnvKeyAutopilotCASecretName
+	DefaultCASecretName = "portworx-api-root-ca"
```

Comment:
In case of operator generated certs, would we be generating just server key and certs? Who provides the root CA?
If all of them are generated by the operator and only portworx clients are going to consume them, why not store then in the same secret.
If we store them in separate secrets, we will have to take two secret names at the top level - 
```
spec:
  tls:
    enabled: bool
    rootCASecret: string
    serverCertSecret: string
    advancedOptions: {}
```
Is this how we would do it?

Reply:
> In case of operator generated certs, would we be generating just server key and certs? Who provides the root CA?
> If all of them are generated by the operator and only portworx clients are going to consume them, why not store then in the same secret.
> If we store them in separate secrets, we will have to take two secret names at the top level -
> 
> ```
> spec:
>   tls:
>     enabled: bool
>     rootCASecret: string
>     serverCertSecret: string
>     advancedOptions: {}
> ```
> 
> Is this how we would do it?

For now, we assume client has generated everything, and placed cert files on the nodes ... therefore advancedOptions does not support cert as the source instead of file (yet). We can discuss this question in a later PR.

As for consuming the certs, there are the following certs:
- The CA certificate (which needs to be consumed by porx (it acts as both server and client for secure node-node communications over TLS) as well as clients like autopilot, stork.
- The Server cert (public key) and Server key (private key) pair that are needed only by porx (server).



What I had in mind for that future PR was this:
```
spec:
  tls:
    enabled: bool
    advancedOptions: {}
```
The ca cert is assumed to be in a "default" secret, so is the server cert and key.


```
spec:
  tls:
    enabled: bool
    advancedOptions:
      serverKey:
	    secretKeyRef:
	      name:
	      key:
      serverCert:
	    secretKeyRef:
	      name:
	      key:

```
`serverKey` are `serverCert` are not in the "default" secret ... so we provide reference to the secret

Reply:
I am not completely clear on how we will be simplifying the workflow for operator generated certs. 
What you have done is this PR is good and we don't need to block this PR as we will be adding changes incrementally. But let's discuss offline on how we plan to simplify TLS experience for most users.